app/gui/controller/engine-protocol/src/language_server/types.rs

```diff
@@ -1155,6 +1155,72 @@ pub struct LibraryComponentGroup {
 }
 
 
+
+// =============================
+// === Execution Environment ===
+// =============================
+
+/// The execution environment which controls the global execution of functions with side effects.
+///
+/// For more information, see
+/// https://github.com/enso-org/design/blob/main/epics/basic-libraries/write-action-control/design.md.
+#[derive(Hash, Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize, Display)]
+pub enum ExecutionEnvironment {
+    /// Allows editing the graph, but the `Output` context is disabled, so it prevents accidental
+    /// changes.
+    Design,
+    /// Unrestricted, live editing of data.
+    Live,
```

Comment:
1. I believe we have the strings "design" and "live" hardcoded somewhere in the previous PR.
2. I don't think that hardcoding that as enum is a good option, as these 2 options are temporary and we will allow users defining custom modes. So we should keep the current mode in string-like struct instead.
3. This will also make all of helper functions simpler, such as `list_all_imstrings`.
4. Function `list_all` is similar to a function in previous PR, which was already listing all available execution environments, we should unify these codes.

Reply:
Here they need to be hard-codes as enum, as this is the current LS protocol. The UI elements can handle arbitrary labels. So, would suggest this to be refactored appropriately once custom modes are implemented.